src/audio/plugins/jack.h

```diff
@@ -0,0 +1,127 @@
+/**
+ *  OSM
+ *  Copyright (C) 2022  Adrian Schollmeyer
+
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef AUDIO_JACKPLUGIN_H
+#define AUDIO_JACKPLUGIN_H
+
+#include "../plugin.h"
+#include <QObject>
+#include <functional>
+#include <jack/jack.h>
+#include <jack/types.h>
```

Comment:
It seems unfortunate that the jack headers are included in this header, which in turn is included in client.cpp. It would be better if the external jack header dependency was fully contained in plugins/jack.cpp . That seems feasible.

Reply:
I wouldn't consider this a big deal. The JACK headers are lightweight and the headers aren't provided as part of a library for other software, anyway. There is no big advantage keeping them away from the plugin's header file and keeping them contained in jack.cpp brings the additional issue of moving the class definition for JackClient to the source file, which is opposed to the code style from the ALSA plugin (which was my template for developing this plugin).